src/Cardano/Crypto/Wallet.hs

```diff
@@ -41,32 +43,43 @@ module Cardano.Crypto.Wallet
     , verify
     ) where
 
-import           Crypto.OpenSSL.Random (randBytes)
-import qualified Crypto.PubKey.Ed25519 as Ed25519
+import           Control.DeepSeq         (NFData)
 import qualified Crypto.ECC.Edwards25519 as Edwards25519
-import           Crypto.Hash (SHA512, hash)
-import qualified Crypto.MAC.HMAC as HMAC
-import           Crypto.Error (throwCryptoError)
+import           Crypto.Error            (throwCryptoError)
+import           Crypto.Hash             (SHA512, hash)
+import qualified Crypto.MAC.HMAC         as HMAC
+import           Crypto.OpenSSL.Random   (randBytes)
+import qualified Crypto.PubKey.Ed25519   as Ed25519
+import           Data.ByteArray          (ByteArrayAccess, convert)
+import qualified Data.ByteArray          as B (append)
+import           Data.ByteString         (ByteString)
+import qualified Data.ByteString         as B (length, pack, splitAt)
+import           Data.Hashable           (Hashable)
 import           Data.Word
-import           Data.ByteString (ByteString)
-import qualified Data.ByteString as B (splitAt, length, pack)
-import           Data.ByteArray (ByteArrayAccess, convert)
-import qualified Data.ByteArray as B (append)
+import           GHC.Generics            (Generic)
 
-import Debug.Trace
+import           Debug.Trace
 
 newtype ChainCode = ChainCode ByteString
-    deriving (Show,Eq)
+    deriving (Show, Eq, Ord, NFData, Hashable)
 
 data XPrv = XPrv !Edwards25519.Scalar !ChainCode
+    deriving (Eq, Ord, Generic)
```

Comment:
this open security issue, so I'm less than keen to derive those classes

The right question, is why do you need generic, eq, ord ? and what do you need XPrv to be able to do ?